src/receivers/HTTPModuleFunctions.ts

```diff
@@ -190,6 +190,13 @@ export class HTTPModuleFunctions {
     args: ReceiverProcessEventErrorHandlerArgs,
   ): Promise<boolean> {
     const { error, response, logger, storedResponse } = args;
+
+    // Check if the response headers have already been sent
+    if (response.headersSent) {
+      logger.error('Headers already sent, cannot send another response');
```

Comment:
Can you adjust the logging for consistency with the following logic?
```suggestion
      logger.error('An unhandled error occurred after ack() called in a listener');
      logger.debug(`Error details: ${error}, storedResponse: ${storedResponse}`);
```